docs/diagrams/FindSequenceDiagram.puml

```diff
@@ -0,0 +1,93 @@
+@startuml
+!include style.puml
+skinparam ArrowFontStyle plain
+
+box Logic LOGIC_COLOR_T1
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":AddressBookParser" as AddressBookParser LOGIC_COLOR
+participant ":FindCommandParser" as FindCommandParser LOGIC_COLOR
+participant "namePredicate:NameContainsKeywordsPredicate" as NameContainsKeywordsPredicate LOGIC_COLOR
+participant "combinedPredicate:FindCommandPredicate" as FindCommandPredicate LOGIC_COLOR
+participant "f:FindCommand" as FindCommand LOGIC_COLOR
+participant "r:CommandResult" as CommandResult LOGIC_COLOR
+
+end box
+
+box Model MODEL_COLOR_T1
+participant "m:Model" as Model MODEL_COLOR
+end box
+
+[-> LogicManager : execute("find n/Alex")
+activate LogicManager
+
+LogicManager -> AddressBookParser : parseCommand("find n/Alex")
+activate AddressBookParser
+
+create FindCommandParser
+AddressBookParser -> FindCommandParser
+activate FindCommandParser
+
+FindCommandParser --> AddressBookParser
+deactivate FindCommandParser
+
+AddressBookParser -> FindCommandParser : parse("n/Alex")
+activate FindCommandParser
+
+create NameContainsKeywordsPredicate
+FindCommandParser -> NameContainsKeywordsPredicate
+activate NameContainsKeywordsPredicate
+
+NameContainsKeywordsPredicate -->FindCommandParser
+deactivate
+
+create FindCommandPredicate
+FindCommandParser -> FindCommandPredicate
+activate FindCommandPredicate
+
+FindCommandPredicate --> FindCommandParser
+deactivate
+
+FindCommandParser -> FindCommandPredicate : addNameContainsKeywordsPredicate(namePredicate)
+activate FindCommandPredicate
+
+FindCommandPredicate --> FindCommandParser
+deactivate FindCommandPredicate
+
+create FindCommand
+FindCommandParser -> FindCommand : new FindCommand(combinedPredicate)
+activate FindCommand
+
+FindCommand --> FindCommandParser :
+deactivate FindCommand
+
+FindCommandParser --> AddressBookParser : f
+deactivate FindCommandParser
+'Hidden arrow to position the destroy marker below the end of the activation bar.
+FindCommandParser -[hidden]-> AddressBookParser
+destroy FindCommandParser
+
+AddressBookParser --> LogicManager : f
+deactivate AddressBookParser
+
+LogicManager -> FindCommand : execute(m)
+activate FindCommand
+
+FindCommand -> Model : updateFilteredPersonList(combinedPredicate)
+activate Model
+
+Model --> FindCommand
+deactivate Model
+
+create CommandResult
+FindCommand -> CommandResult
+activate CommandResult
+
+CommandResult --> FindCommand
+deactivate CommandResult
+
+FindCommand --> LogicManager : r
+deactivate FindCommand
+
+[<--LogicManager
+deactivate LogicManager
+@enduml
```

Comment:
For the diagram, should there be crosses at the end of the lifeline of other classes to demonstrate that those objects are ready for garbage collecting? Additionally values like the names of the object and parameters may be omitted to make the diagram compact and easier to read, but that's more of a personal thing than a syntactic error

Reply:
I don't think there should be crosses at the end of predicate or command class, since command is stored in stack for undo function and predicates should exist as long as command exist

Reply:
I understand, but then should there be one for CommandResult?

Reply:
> I understand, but then should there be one for CommandResult?

ah, hard to say to be honest, because the example deleteSequenceDiagram in the original AB3 developer guide does not mark an x at the end of the CommandResult. There might be some unknown mechanism conserve the CommandResult object. Oh the original developer just made this tiny mistake.